### PR TITLE
CDAP-3632 Adding HTTP Handler for Lineage

### PR DESCRIPTION
**Pull request by poornachandra**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageHandler.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * HttpHandler for lineage.
+ */
+@Path(Constants.Gateway.API_VERSION_3)
+public class LineageHandler extends AbstractHttpHandler {
+  private final LineageService lineageService;
+
+  @Inject
+  public LineageHandler(LineageService lineageService) {
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageHandler.java]: can be package private?

**Comment by bdmogal** [comment]
Minor comment, rest LGTM. Do we need an end point for streams as well?

File: cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java [hunk 2]
```diff
@@ -86,8 +86,8 @@ private String getV3RoutingService(String [] uriParts, AllowedMethod requestMeth
       matches(uriParts, "v3", "namespaces", null, "apps", null, null, null, "metadata", "tags") ||
       matches(uriParts, "v3", "namespaces", null, "datasets", null, "metadata", "tags") ||
       matches(uriParts, "v3", "namespaces", null, "streams", null, "metadata", "tags") ||
-      matches(uriParts, "v3", "namespaces", null, "metadata", "search")) {
-      // all metadata REST APIs are currently exposed from Dataset service
+      matches(uriParts, "v3", "namespaces", null, "metadata", "search") ||
+      matches(uriParts, "v3", "namespaces", null, "datasets", null, "lineage")) {
```
Comment (bdmogal) [inline cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java]: we should add a verification in RouterPathTest

**Comment by poornachandra** [comment]
I'll add endpoint when in a later PR, right now I don't have method to compute lineage for a stream.

File: cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java (hunk 2 above)
Comment (poornachandra) [inline cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java]: Done

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-DUT2827-2

**Comment by bdmogal** [comment]
LGTM. merge after build passes.

**Comment by poornachandra** [comment]
Squashing commits and merging.

**Pull request merged**